### PR TITLE
Port sav_chatot/sound_chatot from pokediamond

### PR DESCRIPTION
**Pull request by Admiral-Fish**

### CODE REVIEW

File: src/sav_chatot.c [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+SOUND_CHATOT *Chatot_new(HeapID heapID) {
+    SOUND_CHATOT *chatot = AllocFromHeap(heapID, sizeof(SOUND_CHATOT));
+    Sav2_Chatot_init(chatot);
+    return chatot;
+}
+
+SOUND_CHATOT *Sav2_Chatot_get(SAVEDATA *saveData) {
+    return SavArray_get(saveData, 0x12);
```
Comment (PikalaxALT) [inline src/sav_chatot.c:20]:
```suggestion
    return SavArray_get(saveData, SAVE_CHATOT);
```

File: src/sav_chatot.c [hunk 2]
```diff
@@ -0,0 +1,82 @@
+#include "sav_chatot.h"
+#include <nitro/mi/memory.h>
+
+u32 Sav2_Chatot_sizeof(void) {
+    return sizeof(SOUND_CHATOT);
+}
+
+void Sav2_Chatot_init(SOUND_CHATOT *chatot) {
+    MI_CpuFill32(chatot, 0, sizeof(SOUND_CHATOT));
```
Comment (PikalaxALT) [inline src/sav_chatot.c:9]:
```suggestion
    MI_CpuClear32(chatot, sizeof(SOUND_CHATOT));
```

File: src/sound_chatot.c [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    if (*r5 == 1) {
+        sub_020058B8(14);
+        sub_02005680(14);
+    }
+
+    *r4 = 0;
+}
+
+u32 Chatot_startRecording(void) {
+    MIC_SamplingData st0;
```
Comment (PikalaxALT) [inline src/sound_chatot.c:96]: Let's copy the `MICAutoParam` data type from NitroSDK and use that instead.

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: src/sound_chatot.c (hunk 3 above)
Comment (Admiral-Fish) [reply src/sound_chatot.c:96]: Should I copy over the enum names and struct field names directly as they are in the SDK?

Comment (PikalaxALT) [reply src/sound_chatot.c:96]: Yes. Be sure to reformat it to be consistent with the style of the rest of the repo too. The definitions go in <lib/include/nitro/spi/mic.h>

File: src/sound_chatot.c [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    switch (a0) {
+        case 0:
+        case 1:
+        case 5:
+        case 11:
+        case 12:
+            return TRUE;
+        default:
+            return FALSE;
+    }
```
Comment (PikalaxALT) [inline src/sound_chatot.c:186]: `case` should be at the same indentation level as `switch`

File: src/sound_chatot.c [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+u32 Chatot_startRecording(void) {
+    MICAutoParam st0;
+    
+    st0.type = MIC_SAMPLING_TYPE_SIGNED_8BIT;
+    st0.buffer = sub_020059D8();
+    st0.size = 2000;
+    if ((st0.size & 0x1f) != 0) {
+        st0.size &= ~0x1f;
+    }
+    st0.rate = 0x4174;
```
Comment (PikalaxALT) [inline src/sound_chatot.c:104]: HW_CPU_CLOCK_ARM7 / 2000

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: src/sound_chatot.c [hunk 6]
```diff
@@ -0,0 +1,187 @@
+#include "sound_chatot.h"
+#include "sound.h"
+#include "constants/species.h"
+#include "math_util.h"
+#include "unk_02004A44.h"
+#include <nitro/spi/ARM9/mic.h>
```
Comment (PikalaxALT) [inline src/sound_chatot.c:6]: This is already included in <nitro/spi.h>, so is not necessary here. (the way this project is set up, all the nitro sdk and nitrosystem headers are recursively included via command line argument)

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

**Review by PikalaxALT — APPROVED** [review]

**Review by tgsm — APPROVED** [review]

**Pull request merged**